projects/components/src/navigation/navigation-list.component.ts

```diff
@@ -1,9 +1,18 @@
 import { ChangeDetectionStrategy, Component, EventEmitter, Input, OnChanges, Output } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
 import { IconType } from '@hypertrace/assets-library';
-import { NavigationService } from '@hypertrace/common';
-import { Observable } from 'rxjs';
-import { map, startWith } from 'rxjs/operators';
+import {
```

Comment:
This is where the time range for each page gets applied on navigation. If both FFs are enabled, decorate and use those navItems, otherwise, use original nav items. The decorated navItems are passed down to the `NavItemComponent`, when one of them is selected, this component handles the event and uses the time range on the passed navItem object to set the time range for the app.

If **_either_** TR or NavigationRedesign are disabled, the nav list and nav items behave like normal.